content/en/admin/config.md

```diff
@@ -202,7 +202,17 @@ Determines the amount of logs generated by Mastodon. Defaults to `info`, which g
 
 #### `TRUSTED_PROXY_IP`
 
-If your Mastodon web process is on the same machine as your reverse proxy (e.g. nginx), then you don't need this setting. Otherwise, you need to set it to the IP from which your reverse proxy sends requests to Mastodon's web process, otherwise Mastodon will record the reverse proxy's own IP as the IP of all requests, which would be bad because IP addresses are used for important rate limits and security functions.
+Tells the Mastodon web and streaming processes which IPs act as your trusted reverse proxy (e.g. nginx). It affects how Mastodon determines the source IP of each request, which is used for important rate limits and security functions. If the value is set incorrectly then Mastodon could use the IP of the reverse proxy instead of the actual source.
+
+By default the loopback and private network address ranges are trusted. Specifically:
+ * `127.0.0.1/8`
+ * `::1/128`
+ * `10.0.0.0/8`
+ * `172.16.0.0/12`
+ * `192.168.0.0/16`
+ * `fc00::/7`
+
+If your Mastodon web and streaming processes run on the same machine or in the same private network as your reverse proxy then you most likely don't need to modify this setting and can use the default. Otherwise it should be set to the IP of the reverse proxy, as a comma-separated list of IPs or IP ranges using [CIDR notation](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing#CIDR_notation).
```

Comment:
This first sentence isn't entirely true because you can often end up in a situation with multiple reverse proxies.

The common situation being if you're using CloudFlare in front of nginx, since CloudFlare IPs are not in private network range.

Reply:
Good point! I updated the wording to try to make it less likely that people would read this and do the wrong thing. It's pretty wordy now, though... Not sure if it's _too_ wordy. I'm definitely open to suggestions.

But also, I don't know, I think I would have appreciated the verbosity when I was configuring my server.

Reply:
Well that's also a good point and the confusing behavior is certainly confusing. I'm wondering if it actually isn't a problem for Mastodon. Is this the relevant code:?

https://github.com/mastodon/mastodon/blob/d11d15748cd24342a133221edb3022022b585119/config/environments/production.rb#L44-L45

```ruby
  # Allow to specify public IP of reverse proxy if it's needed
  config.action_dispatch.trusted_proxies = ENV['TRUSTED_PROXY_IP'].split(/(?:\s*,\s*|\s+)/).map { |item| IPAddr.new(item) } if ENV['TRUSTED_PROXY_IP'].present?
```

So it splits the string into an array of strings then maps them to IPAddrs. So it seems like it will always pass an enumerable and we should get the desirable "replace" behavior rather than the undesirable "append to the defaults" behavior. But again I don't know Ruby and I haven't looked at this super closely. Specifically I have no idea how setting `config.action_dispatch.trusted_proxies` in the Mastodon code results in that value being passed to [this initialize function](https://github.com/rails/rails/blob/ec782106553202e7cf2aeff7cbbce94d85b75635/actionpack/lib/action_dispatch/middleware/remote_ip.rb#L59). It seems like that's a thing that's happening but 🤷

Reply:
Right, I think you're correct. I was trying to work out why an admin recently hadn't needed to include any local addresses in their setup with CloudFlare in k8s, but I think I worked it out.